### PR TITLE
Some font fixes/compatibility improvements 2

### PR DESCRIPTION
**Pull request by kakurasan**
* Use U+FFFD(�) if glyph not found
* U+30FC(ー): Adjust shape for better font compatibility
* Text input: Replace U+301C(〜) with U+FF5E(～) for (font and output) compatibility

9e418df:
9e418df
e070300:
e070300

### CODE REVIEW

**Comment by carstene1ns** [comment]
This looks really good again! 👍
Do you plan to work on other font related things the next days? Then we can keep this open for a while.

File: src/font.cpp [hunk 1]
```diff
@@ -50,7 +50,7 @@ namespace {
 		if(ret != (data + size) && ret->code == code) {
 			return ret;
 		} else {
-			static ShinonomeGlyph const empty_glyph = { 0, true, {0} };
+			static ShinonomeGlyph const empty_glyph = { 65533, true, { 96, 240, 504, 924, 1902, 3967, 4031, 1982, 1020, 440, 240, 96 } };
```
Comment (carstene1ns) [inline src/font.cpp]: Small suggestion: Maybe rename `empty_glyph` to `replacement_glyph`, because it is not empty anymore.

**Comment by kakurasan** [comment]
> Do you plan to work on other font related things the next days?

No, there's nothing to do (font related things) for now.
I plan to open another pull request about RPG2000 battle messages.

**Comment by carstene1ns** [comment]
(For reference: This fixed another part of #514.)

**Comment by Ghabry** [comment]
Is the RPG2k battle message stuff related to the English RPG2k release that had substituion characters for terms? (%S & others)
I have started working on this in a branch, if you want a start: https://github.com/Ghabry/easyrpg-player/commit/309f5e5f0992700978a80c7bf4cabdd4403e8839

Well, the substitution handling is not really in there, only the engine detection. And regex can't be used because it's missing on Android and other platforms (there is a Tokenizer in Utils instead)

**Comment by Ghabry** [comment]
btw we forgot to PR build test his code ;)

**Comment by kakurasan** [comment]
> Is the RPG2k battle message stuff related to the English RPG2k release (version 1.61) that had substituion characters for terms? (%S & others)

No, mainly the wrong uses of Data::terms.* (e.g. Data::terms.actor_critical &lt;--&gt; Data::terms.enemy_critical) in game_battlealgorithm.cpp.

https://github.com/EasyRPG/Player/compare/master...kakurasan:2k_battle_message

**Comment by Ghabry** [comment]
btw thanks for these nice screenshots. Because we always prepare a blog post containing screenshots and videos of changes since the previous release they are ideal. :+1:

**Pull request merged**